autogen/trace/trace.py

```diff
@@ -1,34 +1,34 @@
 from typing import Optional, List, Dict, Callable, Union, Type, Any, Tuple
-from autogen.trace.nodes import MessageNode, Node, ParameterNode, GRAPH, USED_NODES, NAME_SCOPES
+from autogen.trace.nodes import Node, GRAPH
+from autogen.trace.utils import for_all_methods
 from autogen.agentchat.agent import Agent
 from autogen.agentchat.conversable_agent import ConversableAgent
-import inspect
-from dill.source import getsource
+from autogen.trace.trace_autogen import trace_ConversableAgent
 
-def node(message):
-    # wrap the message as a Node if it's not
-    if isinstance(message, Node):
-        return message
-    return Node(message)
 
-def trace(fun):
-    if inspect.isclass(fun):
-        if issubclass(fun, ConversableAgent):
-            return trace_ConversableAgent(fun)
-    # it should be a function
-    return trace_operator(fun)
+## These are external APIs.
+
+from autogen.trace.nodes import node
+""" Create a Node from a message. If message is already a Node, return it.
+    This method is for the convenience of the user, it should be used over
+    directly invoking Node."""
+
+
+def trace(cls):
+    """ A decorator to trace a ConversableAgent."""
+    assert issubclass(cls, ConversableAgent), "cls must be a subclass of ConversableAgent."
+    return trace_ConversableAgent(cls)
+    # TODO: enable tracing other classes and functions
 
 class no_trace():
```

Comment:
Is the intended use of this `no_trace(agent)`? Then shouldn't we have an `__init__` method that takes in the agent? Also, since the name is really close to `trace(agent)` -- it should have the same input signature.

Reply:
Good point. The original intended usage is a context manager, which acts like `with torch.no_grad():`. I see it has a confusing name as `trace(agent)`

Reply:
Oh I see, `no_trace` is used as `with no_trace():` in `compatability` -- I'll still raise the same usability/design point before...might be nice to make `trace()` and `no_trace()` behave the same, and introduce a different context manager like `autogen.trace.no_trace_mode():`

Reply:
Yes, we should change the current naming of APIs. They are confusing.